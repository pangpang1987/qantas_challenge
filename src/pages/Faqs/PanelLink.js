import React from "react";
import PropTypes from "prop-types";

import { Link } from "./styled";

class PanelLink extends React.PureComponent {
  handleClick = () => {
    const { index, setActiveIndex } = this.props;
    setActiveIndex(index);
  };

  render() {
    const { index, activeIndex, title } = this.props;
    const isActive = index === activeIndex;
    return (
      <Link className={isActive ? 'isActive' : ''} onClick={this.handleClick}>
        {title}
      </Link>
    );
  }
}

PanelLink.propTypes = {
  title: PropTypes.string,
  index: PropTypes.number,
  activeIndex: PropTypes.number,
  setActiveIndex: PropTypes.func
};

export default PanelLink;
