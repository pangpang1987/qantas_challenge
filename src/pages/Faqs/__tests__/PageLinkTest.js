import React from "react";

import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import PanelLink from "../PanelLink";

describe("Component PanelLink", () => {
  const setActiveIndex = jest.fn();
  it("should render normal state", () => {
    const component = shallow(<PanelLink index={1} activeIndex={0} title="test" setActiveIndex={setActiveIndex} />);
    const tree = toJson(component);
    expect(tree).toMatchSnapshot();
  });

  it("should render active state", () => {
    const component = shallow(<PanelLink index={0} activeIndex={0} title="test" setActiveIndex={setActiveIndex} />);
    const tree = toJson(component);
    expect(tree).toMatchSnapshot();
  });

  it("should render active state", () => {
    const component = shallow(<PanelLink index={0} activeIndex={1} title="test" setActiveIndex={setActiveIndex} />);
    component.instance().handleClick();

    expect(setActiveIndex.mock.calls).toEqual([[0]]);
    const tree = toJson(component);
    expect(tree).toMatchSnapshot();
  });
});
