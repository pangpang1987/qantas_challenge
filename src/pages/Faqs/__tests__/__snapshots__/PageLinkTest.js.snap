// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component PanelLink should render active state 1`] = `
<styled.li
  className="isActive"
  onClick={[Function]}
>
  test
</styled.li>
`;

exports[`Component PanelLink should render active state 2`] = `
<styled.li
  className=""
  onClick={[Function]}
>
  test
</styled.li>
`;

exports[`Component PanelLink should render normal state 1`] = `
<styled.li
  className=""
  onClick={[Function]}
>
  test
</styled.li>
`;
